Source/ModelGroup.cs

```diff
@@ -16,6 +16,7 @@ internal abstract partial class ModelGroup
         public List<Runtime.Image> UsedFigures = new List<Runtime.Image>();
         public int Id = -1;
         public bool NoSampleImages = false;
```

Comment:
this should be readonly

Reply:
actually, maybe all the members should be readonly.

Reply:
I'm not quite sure I follow. `NoSampleImages` needs to be `false` for every model group except `Compatibility`. If `NoSampleImages` is readonly I couldn't set it for `Compatibility`. 

I could do something like `public abstract bool ApplyPostRuntimeChanges { get; }` but then I'd need to override it in every model group, even though only `Compatibility` needs the value changed.

Reply:
readonly doesn't mean what you think it does. readonly just means it can't be set outside of the constructor.

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly

Reply:
Something like this?

In Model Group I've added

        `public ModelGroup()
        {
            NoSampleImages = false;
        }

        public ModelGroup(bool dontUseSampleImages)
        {
            NoSampleImages = dontUseSampleImages;
        }`

and in Compatibility

`public Compatibility(List<string> imageList) : base(true)`

Reply:
This isn't necessary. You are setting the flag in the constructor of the child class. You just need to mark the field as readonly.

Reply:
readonly doesn't work that way.

If ModelGroup has
`public readonly bool NoSampleImages;`

and Compatibility has 
`NoSampleImages = true;`

Then an error is displayed for the line in Compatibility "A readonly field cannot be assigned to (Except in a constructor or a variable initializer"

https://stackoverflow.com/questions/6037546/assigning-a-value-to-an-inherited-readonly-field

Reply:
Ahh, that's terrible. So, either do what you did with the base constructor, or make the field into a property like this:
```
public bool NoSampleImages { get; private set; }
```

Reply:
Another way is to make the property virtual and override it to be the correct value. I think I would prefer the base constructor method.

Reply:
Make sure to make the base constructor `protected` and not `public`

Reply:
> I can revert that part if you want

Yes, please.